roles/servicemesh/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+- name: Update current state
+  k8s_info:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: '{{ meta.name }}'
+    namespace: '{{ operator_namespace }}'
+  register: resource_state
+
+- when: resource_state.resources[0].status.state == "Init"
+  block:
+  - name: Check Hyperfoil-status
+    k8s_info:
+      api_version: hyperfoil.io/v1alpha2
+      kind: Hyperfoil
+      name: "{{ meta.name }}-hyperfoil-{{ trunc_uuid }}"
+      namespace: '{{ operator_namespace }}'
+    register: hf_state
+
+  - name: Check ServiceMesh control plane
+    k8s_info:
+      api_version: maistra.io/v1
+      kind: ServiceMeshControlPlane
+      name: '{{ meta.name }}'
+      namespace: "{{ controlplane_namespace }}"
+    register: smcp_state
+
+  - set_fact:
+      smcp_ready: "{{ smcp_state | json_query('resources[0].status.conditions[]') | selectattr('type','match','Ready') | selectattr('status', 'match', 'True') | list | length > 0 }}"
+
+  - name: Progress to deploy
+    when: hf_state.resources[0].status.status == "Ready" and smcp_ready
+    operator_sdk.util.k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Deploy"
+        complete: false
+
+- name: Update current state
+  k8s_info:
+    api_version: ripsaw.cloudbulldozer.io/v1alpha1
+    kind: Benchmark
+    name: '{{ meta.name }}'
+    namespace: '{{ operator_namespace }}'
+  register: resource_state
+
+- name: Check ingress route
+  k8s_info:
+    api_version: route.openshift.io/v1
+    kind: Route
+    name: istio-ingressgateway
+    namespace: '{{ controlplane_namespace }}'
+  register: ingress_route
+
+- name: Set app route
+  when: ingress_route.resources | length > 0
+  set_fact:
+    #TODO: maybe use trunc_uuid in the hostname?
+    app_route: "{{ meta.name }}-app-{{ trunc_uuid}}{{ ingress_route.resources[0].status.ingress[0].host | regex_replace('^[^.]*', '')}}"
+
+- when: resource_state.resources[0].status.state == "Deploy"
+  block:
+  - name: Generate certificates
+    command: openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/app.key -out /tmp/app.crt -subj "/CN={{ app_route }}/O=app"
+
+  - name: Create ingressgateway certs secret
+    k8s:
+      definition:
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          name: istio-ingressgateway-certs
+          namespace: "{{ controlplane_namespace }}"
+          ownerReferences: "{{ owner }}"
+        data:
+          tls.key: "{{ lookup('file', '/tmp/app.key') | b64encode }}"
+          tls.crt: "{{ lookup('file', '/tmp/app.crt') | b64encode }}"
+
+  - name: Get ingress gateway pods
+    k8s_info:
+      api_version: v1
+      kind: Pod
+      namespace: "{{ controlplane_namespace }}"
+      label_selectors:
+      - istio = ingressgateway
+    register: ingressgateways
+
+  - name: Reboot ingress gateway(s)
```

Comment:
I need some background here. Why do you reboot pods?

Reply:
We've just created a new secret that's optionally bound to ingress gw's filesystem. Even if Openshift makes it appear on the filesystem, AFAIK Envoy (the GW) won't pick it up so we need to reboot it. It might be possible to send it some kind of signal, but just deleting the pods is the most reliable way in my experience (these get re-created by the deployment).